numpy/_utils/__init__.py

```diff
@@ -27,3 +29,59 @@ def decorator(func):
             func.__module__ = module
         return func
     return decorator
+
+
+def _rename_parameter(old_names, new_names, dep_version=None):
```

Comment:
Adapted from https://github.com/scipy/scipy/blob/2c6f5c67d5870e3058c8b7b7818bcb8a940bfeee/scipy/_lib/_util.py#L730

I can bring in more extensive tests of the decorator (https://github.com/scipy/scipy/blob/5dc8a79e482118d150d9430598f52d17251317f1/scipy/_lib/tests/test__util.py#L221) if desired.